### ISSUE TITLE
issue adding Canis_lupus_familiaris IGKC and IGKL

### ISSUE THREAD

**Issue by avilella** [snijderlab/stitch#155]
I am struggling to create a template file for the species and constants above. The IGHC are fine, but the IGKC and IGKL give me errors:

>IGKCcf
RNDAQPAVYLFQPSPDQLHTGSASVVCLLNSFYPKDINVKWKVDGVIQDTGIQESVTEQDKDSTYSLSSTLTMSSTEYLSHELYSCEITHKSLPSTLIKSFQRSECQRVD

```
ERROR: SubArray Exception length 1 index 7 on an array of length 0
STACKTRACE:    at AssemblyNameSpace.HelperFunctionality.SubArray[T](T[] data, Int32 index, Int32 length)
   at AssemblyNameSpace.Template.AlignedSequences()
   at AssemblyNameSpace.Template.CombinedSequence()
   at AssemblyNameSpace.Template.ConsensusSequence()
   at AssemblyNameSpace.RunParameters.SingleRun.CreateRecombinationTemplates(IEnumerable`1 combinations, List`1 order, Alphabet alphabet, Segment parent, NameFilter namefilter)
   at AssemblyNameSpace.RunParameters.SingleRun.RunRecombine(List`1 segments, List`1 recombined_segment)
   at AssemblyNameSpace.RunParameters.SingleRun.Calculate()
   at AssemblyNameSpace.ToRunWithCommandLine.RunBatchFile(String filename)
   at AssemblyNameSpace.ToRunWithCommandLine.Main()
SOURCE: stitch
TARGET: T[] SubArray[T](T[], Int32, Int32)
```

**Comment by douweschulte** [comment]
That does not look like a nice friendly error, so that is a bug.
Could you share the full files for IGKC and IGKL? Whit those I can try to fix your issue and create better error messages.

**Comment by avilella** [comment]
>IGKCcf
RNDAQPAVYLFQPSPDQLHTGSASVVCLLNSFYPKDINVKWKVDGVIQDTGIQESVTEQDKDSTYSLSSTLTMSSTEYLSHELYSCEITHKSLPSTLIKSFQRSECQRVD
>IGLC1cf
GQPKSSPLVTLFPPSSEELGANKATLVCLISDFYPSGLKVAWKADGSTIIQGVETTKPSKQSNNKYTASSYLSLTPDKWKSHSSFSCLVTHQGSTVEKKVAPAECS
>IGLC2cf
GQPKASPSVTLFPPSSEELGANKATLVCLISDFYPSGVTVAWKADGSPGIQGVETTKPSKQSNNKYAASSYLSLTPDKWKSHSSFSCLVTHEGSTVEKKVAPAECS
>IGLC3p4p6p7p8cf
GQPKASPSVTLFPPSSEELGANKATLVCLISDFYPSGVTVAWKADGSPVTQGVETTKPSKQSNNKYAASSYLSLTPDKWKSHSSFSCLVTHEGSTVEKKVAPAECS
>IGLC5p9cf
GQPKASPSVTLFPPSSEELGANKATLVCLISDFYPSGVTVAWKADGSPITQGVETTKPSKQSNNKYAASSYLSLTPDKWKSHSSFSCLVTHEGSTVEKKVAPAECS

**Comment by douweschulte** [comment]
The fasta file as shown looks fine, I could not recreate any errors with it. But I saw you are using `Canis lupus familiaris` so I added the data from IMGT to our standard templates, in this way the colouration (CDRs, conserved amino acids, ...) is retained in the output of Stitch. After close examination of the code at the error reported position a possible bug was fixed.

The newest version with this bugfix and the templates for `Canis lupus familiaris` can be found here: https://github.com/snijderlab/stitch/actions/runs/2147601209.

Thanks again for the error report!
